### ISSUE TITLE
您好，请问如何进行TSNE降维可视化？

### ISSUE THREAD

**Issue by YyyZzzCc** [Littleor/MLDSO#1]
您好，请问如何进行TSNE降维可视化？

**Comment by Littleor** [comment]
请参考 `utils/main.py` 中的：

```python
        points = []
        tsne = manifold.TSNE(n_components=point_latitude)
        if torch.cuda.is_available() and support_set.device != torch.device('cpu'):
            p = tsne.fit_transform(query_set_feature.detach().cpu().numpy())  # Size([nq, point_latitude])
            labels = y_hat.view(-1, 1).detach().cpu().numpy()  # Size([nq, 1])
        else:
            p = tsne.fit_transform(query_set_feature.detach().numpy())  # Size([nq, point_latitude])
            labels = y_hat.view(-1, 1).detach().numpy()  # Size([nq, 1])
        points.extend(np.hstack((p, labels)).tolist())
```

通过 `p = tsne.fit_transform(query_set_feature.detach().numpy())` 即可降纬至指定纬度，然后再通过可视化的库将其可视化。

**Comment by YyyZzzCc** [comment]
您好，非常感谢您的回复！请问，labels的作用是什么？
&nbsp;


梦里梦见她梦他
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年10月17日(星期二) 下午4:49
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [Littleor/MLDSO] 您好，请问如何进行TSNE降维可视化？ (Issue #1)





 
请参考 utils/main.py 中的：
         points = []         tsne = manifold.TSNE(n_components=point_latitude)         if torch.cuda.is_available() and support_set.device != torch.device('cpu'):             p = tsne.fit_transform(query_set_feature.detach().cpu().numpy())  # Size([nq, point_latitude])             labels = y_hat.view(-1, 1).detach().cpu().numpy()  # Size([nq, 1])         else:             p = tsne.fit_transform(query_set_feature.detach().numpy())  # Size([nq, point_latitude])             labels = y_hat.view(-1, 1).detach().numpy()  # Size([nq, 1])         points.extend(np.hstack((p, labels)).tolist())
 
通过 p = tsne.fit_transform(query_set_feature.detach().numpy()) 即可降纬至指定纬度，然后再通过可视化的库将其可视化。

**Comment by Littleor** [comment]
> 您好，非常感谢您的回复！请问，labels的作用是什么？

这里的 `labels` 是标签，代表故障分类。
